src/components/TaskDashboard.tsx

```diff
@@ -31,8 +32,13 @@ function TaskDashboard({ taskList }: TaskDashboardProps) {
         className="todo-list stack-large stack-exception"
         aria-labelledby="list-heading"
       >
-        {taskList.map((task) => (
-          <TaskItem key={task.id} taskName={task.value} />
+        {taskList.map((task, index) => (
+          <TaskItem
+            key={task.id}
+            taskName={task.value}
+            index={index}
+            deleteTask={deleteTask}
```

Comment:
Que tal juntar essas linhas e evitar passar o index?
```suggestion
            deleteTask={() => deleteTask(index)}
```